src/ray/gcs/redis_context.cc

```diff
@@ -306,14 +310,20 @@ Status AuthenticateRedis(redisAsyncContext *context, const std::string &password
 
 void RedisAsyncContextDisconnectCallback(const redisAsyncContext *context, int status) {
   RAY_LOG(DEBUG) << "Redis async context disconnected. Status: " << status;
-  // Reset raw 'redisAsyncContext' to nullptr because hiredis will release this context.
-  reinterpret_cast<RedisAsyncContext *>(context->data)->ResetRawRedisAsyncContext();
```

Comment:
Do we need to change this, I thought we do Reconnect during `RedisResponseFn`

Reply:
I saw from the README of `hiredis`, and it says that when a reconnect is needed,
the disconnect callback is a good point to do so.

https://github.com/redis/hiredis/blob/8d8703ee6149ffd3b38cebedd71ad8d1fac63cc7/README.md?plain=1#L454-L455

Reply:
Could you check which thread will call the disconnect callback?

Reply:
Also I think we can lazy reconnect during `RedisResponseFn` instead of during the disconnect callback: if there is no request, we don't need to reconnect right way.

Reply:
The thread name is `gcs_server`.